build.rs

```diff
@@ -547,9 +547,13 @@ fn obj_path(out_dir: &Path, src: &Path) -> PathBuf {
 
 fn configure_cc(c: &mut cc::Build, target: &Target, c_root_dir: &Path, include_dir: &Path) {
     let compiler = c.get_compiler();
-    // FIXME: On Windows AArch64 we currently must use Clang to compile C code
-    let compiler = if target.os == WINDOWS && target.arch == AARCH64 && !compiler.is_like_clang() {
-        let _ = c.compiler("clang");
+    // FIXME: On Windows AArch64 we currently must use clang or clang-cl to compile C code
+    let compiler = if target.os == WINDOWS
+        && target.arch == AARCH64
+        && !compiler.is_like_clang()
+        && !compiler.is_like_clang_cl()
```

Comment:
Which is more commonly used, clang-cl or clang? Is there advantage to putting one check in front of the other?

Reply:
We're unlikely to keep this function at all, in favour of a `compiler.family()` and manual `match` or `matches!()` against `cc::ToolFamily`. Given that GNU cannot compile to MSVC at all (and that is caught inside `cc-rs`, though I don't know if this `ring` workaround is supposed to apply to `aarch64-pc-windows-gnu` as well), we might also replace these two lines with `&& compiler.is_like_msvc()` and document that `cl.exe` is unable to compile these files (which I understand to be the original comment)?

Reply:
It does seem like the logic should be "if the compiler is MSVC, then use clang-cl instead."

Reply:
We don't do anything to support -gnu windows targets, especially for aarch64, so you don't need to worry about them.

Reply:
Thanks @mati865; yes the logic in this PR was written to not change anything if the used compiler is already LLVM-based.